proto/store/approval.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package bytebase.store;
+
+option go_package = "generated-go/store";
+
+// IssuePayloadApproval is a part of the payload of an issue.
+// IssuePayloadApproval records the approval template used and the approval history.
+message IssuePayloadApproval {
+  repeated ApprovalTemplate approval_template = 1;
+  repeated ApprovalHistory history = 2;
+}
+
+message ApprovalHistory {
+  // The `uid` of the approval node.
+  string node_uid = 1;
+
+  enum NodeStatus {
+    NODE_STATUS_UNSPECIFIED = 0;
+    PENDING = 1;
+    APPROVED = 2;
+  }
+  // The new status.
+  NodeStatus status = 2;
+
+  // The principal id of the approver.
+  int32 creator_id = 3;
+}
+
+message ApprovalTemplate {
+  ApprovalFlow flow = 1;
+  string title = 2;
+  string description = 3;
+  int32 creator_id = 4;
+}
+
+message ApprovalFlow {
+  repeated ApprovalStep steps = 1;
+}
+
+message ApprovalStep {
+  // Type of the ApprovalStep
+  // ALL means every node must be approved to proceed.
+  // ANY means approving any node will proceed.
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    ALL = 1;
+    ANY = 2;
+  }
+  Type type = 1;
+
+  repeated ApprovalNode nodes = 2;
+}
+
+message ApprovalNode {
+  // uid uniquely identifies a node in a flow.
+  string uid = 1;
+
+  // Type of the ApprovalNode.
+  // type determines who should approve this node.
+  // ANY_IN_GROUP means the ApprovalNode can be approved by an user from our predefined user group.
+  // See GroupValue below for the predefined user groups.
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    ANY_IN_GROUP = 1;
+  }
+  Type type = 2;
+
+  // GroupValue is used if ApprovalNode Type is ANY_IN_GROUP
+  // The predefined user groups are:
+  // - WORKSPACE_OWNER
+  // - DBA
+  // - PROJECT_OWNER
+  // - PROJECT_MEMBER
+  enum GroupValue {
+    GROUP_VALUE_UNSPECIFILED = 0;
+    WORKSPACE_OWNER = 1;
+    DBA = 2;
```

Comment:
WORKSPACE_DBA

Because later, we will have a project-level role group called maybe something like PROJECT_ASSIGNED_DBA

Reply:
done